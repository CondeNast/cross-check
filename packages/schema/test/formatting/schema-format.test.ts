import { subject, setupSchemaTest, teardownSchemaTest } from "../support";

describe("[schema] formatting - schemaFormat", () => {
  beforeEach(setupSchemaTest);
  afterEach(teardownSchemaTest);

  test("simple", () => {
    let { format } = subject();
    expect(
      format.published.schemaFormat("SimpleArticle"),
    ).toMatchSnapshot();
  });

  test("detailed - published", () => {
    let { format } = subject();
    expect(
      format.published.schemaFormat("MediumArticle"),
    ).toMatchSnapshot();
  });

  test("relationships", () => {
    let { format } = subject();
    expect(
      format.published.schemaFormat("Related"),
    ).toMatchSnapshot();
  });
});
