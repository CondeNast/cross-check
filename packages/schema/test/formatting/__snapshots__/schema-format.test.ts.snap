// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[schema] formatting - schemaFormat detailed - published 1`] = `
"Record(\\"MediumArticle\\", {
  hed: SingleLine().required(),
  dek: Text(),
  body: Text().required(),
  author: Dictionary({
    first: SingleLine(),
    last: SingleLine()
  }),
  issueDate: ISODate(),
  canonicalUrl: Url(),
  tags: List(SingleWord()),
  categories: List(SingleLine()).required(),
  geo: Dictionary({
    lat: Integer().required(),
    long: Integer().required()
  }),
  contributors: List(Dictionary({
    first: SingleLine(),
    last: SingleLine()
  })),
  relatedArticles: hasMany(MediumArticle)
}).metadata({
  collectionName: \\"medium-articles\\",
  modelName: \\"medium-article\\"
})"
`;

exports[`[schema] formatting - schemaFormat relationships 1`] = `
"Record(\\"Related\\", {
  first: SingleLine(),
  last: Text(),
  person: hasOne(SimpleArticle).required(),
  articles: hasMany(MediumArticle)
}).metadata({
  collectionName: \\"related-articles\\",
  modelName: \\"related-article\\"
})"
`;

exports[`[schema] formatting - schemaFormat simple 1`] = `
"Record(\\"SimpleArticle\\", {
  hed: SingleLine().required(),
  dek: Text(),
  body: Text().required()
}).metadata({
  collectionName: \\"simple-articles\\",
  modelName: \\"simple-article\\"
})"
`;
